react/provider/providerLists.tsx

```diff
@@ -130,56 +119,44 @@ const ProviderLists: FC = (props) => {
 
   const statusState = useFilterState({
     items: statusOptions,
-    onChange: ({ selected }) => changeFilterStatus(selected),
+    onChange: ({ selected }) => setSelectedStatus(selected),
     label: formatMessage(filterStatus.status),
   })
 
-  function changeFilterStatus(selected: string | null) {
-    searchListRawQuery({
-      variables: {
-        page: pagination.currentPage,
-        pageSize: ITEMS_PER_PAGE,
-        filter: {
-          ownerEmail: searchEmailFilter,
-          eventDateRange: selected ? dictionaryStatus[selected] : null,
-        },
-      },
-    })
-  }
+  useEffect(() => {
+    const filter = {
+      name: debouncedValue ?? null,
+      ownerEmail: searchEmailFilter ?? null,
+      eventDateRange: selectedStatus ? dictionaryStatus[selectedStatus] : null,
+      createdDateRange: selectedDate ? dictionaryDate[selectedDate] : null,
+    }
 
-  function changeFilterDate(selected: string | null) {
-    if (selected === '9') modalState.setVisible(true)
     searchListRawQuery({
       variables: {
+        filter,
         page: pagination.currentPage,
         pageSize: ITEMS_PER_PAGE,
-        filter: {
-          ownerEmail: searchEmailFilter,
-          createdDateRange: selected ? dictionaryDate[selected] : null,
-        },
       },
     })
-  }
+  }, [
+    debouncedValue,
+    selectedDate,
+    pagination.currentPage,
+    selectedStatus,
+    infoUserList?.ownerEmail,
+    searchEmailFilter,
+  ])
 
   function salveDatePersonalizate() {
+    dictionaryDate[9] = {
```

Comment:
The `dictionaryDate` is not a constant value? Why do you are changing ir here?

Reply:
Então, o 9 seria o personalizado, mas quando a pessoa escolhe as data eu atualizo o valor no dictionary
Para que depois eu posso usar no filtro dessa forma:       createdDateRange: selectedDate ? dictionaryDate[selectedDate] : null,

Reply:
Se esse valor vai ser usado somente dentro desse contexto, acredito que seria melhor criar um state pra guardar esse valor, o que acha?
Mudar um valor que é uma constante que pode ser acessada dentro de outros lugares do código pode gerar problemas no futuro, caso alguém use essa informação pensando ser um valor estático, mas está sendo variável nesse caso.

Reply:
Um outro ponto de melhoria também seria nomear os índices ao invés de numerar eles com 1,2,3,4...
Colocando valores nomeados ficaria mais fácil de identificar o que cada índice irá retornar.
No caso do índice 9, poderia ser algo do tipo `dictionaryDate['customDate']`.
Mas só uma sugestão para implementações futuras, pra dar mais legibilidade ao código 🤓

Reply:
> 

Na verdade essas valores ja estao sendo guardados em um state tambem
Mas achei que guardando no dicionário seria a melhor forma de deixar armazenada, para que em outro momento eu possa fazer uma busca no banco de dados 
Aonde eu pego os valores do dicionario para a pesquisa, sendo a data personalizada ou data fixa
Assim a forma de acessar o dados se manteria uniforme, já que sempre buscaria pelos campos do dicionario. Em vez de realizar  varias verificações, como usar os dados do dicionario ou buscar dados de outras variaveis
Não sei se consegui explicar bem,  mas qualquer duvida me manda que eu tento explicar melhor :)

Reply:
@wagnerlduarte Como essa implementação tem que subir até segunda estarei aprovando ela aqui, mas me comprometo abrir um PR para fazer as mudanças sugeridas!